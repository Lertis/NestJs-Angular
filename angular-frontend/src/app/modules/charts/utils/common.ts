export function dynamicChartLabel(length = 5): string {
	let result = '';
	const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
	const charactersLength = characters.length;
	for (let i = 0; i < length; i++) {
		result += characters.charAt(Math.floor(Math.random() * charactersLength));
	}
	return result;
}

export function randomNumber(min: number, max: number): number {
	if (min > max) {
		[min, max] = [max, min];
	}
	return Math.floor(Math.random() * (max - min) + min);
}
