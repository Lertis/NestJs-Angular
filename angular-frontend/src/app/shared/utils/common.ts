import { each } from 'lodash';

export function fieldValuesChanged(current, original) {
	const keysCollection = [];
	const changed: boolean[] = [];

	each(current, (val, key) => {
		keysCollection.push(key);
	});

	keysCollection.forEach((el) => {
		changed.push(current[el] === original[el]);
	});

	return changed;
}
