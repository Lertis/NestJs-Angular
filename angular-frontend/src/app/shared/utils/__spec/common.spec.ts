import { FORMS_COMPARE } from './common.set';
import { fieldValuesChanged } from '../common';

declare const expect: jest.Expect;

describe('[Function]: fieldValuesChanged', () => {
	FORMS_COMPARE.forEach((test) => {
		it(test.testName, () => {
			const result = fieldValuesChanged(test.current, test.original);
			expect(result).toMatchSnapshot();
		});
	});
});
