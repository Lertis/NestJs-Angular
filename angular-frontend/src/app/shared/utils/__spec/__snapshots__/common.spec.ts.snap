// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Function]: fieldValuesChanged 1 1`] = `
Array [
  true,
  false,
  true,
]
`;

exports[`[Function]: fieldValuesChanged 2 1`] = `
Array [
  false,
  false,
  true,
]
`;

exports[`[Function]: fieldValuesChanged 3 1`] = `
Array [
  true,
  false,
  false,
]
`;

exports[`[Function]: fieldValuesChanged 4 1`] = `
Array [
  false,
  false,
  true,
]
`;

exports[`[Function]: fieldValuesChanged 5 1`] = `
Array [
  false,
  false,
  false,
]
`;

exports[`[Function]: fieldValuesChanged 6 1`] = `
Array [
  false,
  false,
  false,
]
`;
